inc/zoo/swar/SWAR.h

```diff
@@ -231,6 +263,12 @@ template<int NBits, typename T>
 struct BooleanSWAR: SWAR<NBits, T> {
     using Base = SWAR<NBits, T>;
 
+    template <std::size_t N, typename = std::enable_if_t<N == Base::Lanes, T>>
+    constexpr BooleanSWAR(Literals_t<NBits, T>, const bool (&values)[N]) : Base{0} {
+        constexpr auto msbOfFirstLane = T{1} << (NBits - 1);
+        this->m_v = Base::loadIntoLanes(values, [](auto x) { return x ? msbOfFirstLane : 0; });
+    }
```

Comment:
This contains the same antipattern.
A `BooleanSWAR` is a SWAR formally of the same base type but whose lanes are a boolean in the most significant bit, the code ought to reflect that.
You did not indicate a response to my comment that this suffices:
`: Base(Literals<NBits, T>, values) { this->m_v << (NBits - 1); }`

Reply:
oooh damn, i see... yeah that works

Reply:
Ok I don't understand why this would work? Doesn't seem to produce the correct result. Please feel free to push a change to this branch with something that you think works better.